app/src/ble.c

```diff
@@ -280,9 +274,34 @@ char *zmk_ble_active_profile_name() { return profiles[active_profile].name; }
 
 #if IS_ENABLED(CONFIG_ZMK_SPLIT_ROLE_CENTRAL)
 
-void zmk_ble_set_peripheral_addr(bt_addr_le_t *addr) {
-    memcpy(&peripheral_addr, addr, sizeof(bt_addr_le_t));
-    settings_save_one("ble/peripheral_address", addr, sizeof(bt_addr_le_t));
+int zmk_ble_put_peripheral_addr(const bt_addr_le_t *addr) {
+    for (int i = 0; i < ZMK_BLE_SPLIT_PERIPHERAL_COUNT; i++) {
+        // If the address is recognized and already stored in settings, return
+        // index and no additional action is necessary.
+        if (!bt_addr_le_cmp(&peripheral_addrs[i], addr)) {
+            return i;
+        }
```

Comment:
What if the recorded peripheral appears after slot `i+1`, and takes `i`th slot?

Reply:
I don't quite understand what `i+1` you are referring to.

Reply:
I wonder if the `addr` is actually recorded at some slot `j`, where `j` > `i`, but somehow the `i`th slot is open, so the `addr` takes away the `i`th slot as well.

This may be a rare case, and `j`th slot anyhow could be recycled for the `addr` should trigger `disconnected` function in normal cases.

Reply:
Slot 0 is filled, then slot 1 is filled, and so on. There is no code to delete just a `ble/peripheral_addresses/%d` setting so it's not possible to have `i` open and `j` occupied where `i < j`.